src/Resources/Icons/IconsProcessor.php

```diff
@@ -4,45 +4,31 @@
 
 namespace Ssch\TYPO3Rector\Resources\Icons;
 
-use Rector\Core\Application\FileSystem\RemovedAndAddedFilesCollector;
-use Rector\Core\Configuration\Configuration;
 use Rector\Core\Contract\Processor\FileProcessorInterface;
-use Rector\Core\Contract\Rector\RectorInterface;
 use Rector\Core\ValueObject\Application\File;
-use Rector\FileSystemRector\ValueObject\AddedFileWithContent;
 use Ssch\TYPO3Rector\Helper\FilesFinder;
-use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
-use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Ssch\TYPO3Rector\Rector\Resources\IconsRector;
 
 /**
  * @changelog https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/8.3/Feature-77349-AdditionalLocationsForExtensionIcons.html
  * @see \Ssch\TYPO3Rector\Tests\Resources\Icons\IconsProcessor\IconsProcessorTest
  */
-final class IconsProcessor implements FileProcessorInterface, RectorInterface
+final class IconsProcessor implements FileProcessorInterface
 {
     /**
-     * @var Configuration
-     */
-    private $configuration;
-
-    /**
-     * @var RemovedAndAddedFilesCollector
+     * @var FilesFinder
      */
-    private $removedAndAddedFilesCollector;
+    private $filesFinder;
 
     /**
-     * @var FilesFinder
+     * @var IconsRector
      */
-    private $filesFinder;
+    private $iconsRector;
 
-    public function __construct(
-        Configuration $configuration,
-        RemovedAndAddedFilesCollector $removedAndAddedFilesCollector,
-        FilesFinder $filesFinder
-    ) {
-        $this->configuration = $configuration;
-        $this->removedAndAddedFilesCollector = $removedAndAddedFilesCollector;
+    public function __construct(FilesFinder $filesFinder, IconsRector $iconsRector)
+    {
         $this->filesFinder = $filesFinder;
+        $this->iconsRector = $iconsRector;
     }
```

Comment:
Should be array of all Rectors:

```suggestion
    /**
     * @param IconsRector[] $iconRectors
     */
    public function __construct(FilesFinder $filesFinder, array $iconsRectors)
    {
        $this->filesFinder = $filesFinder;
        $this->iconsRectors = $iconsRectors;
    }
```